File: app/src/main/java/com/song/deviceinfo/info/DeviceInfo.java
package com.song.deviceinfo.info;

import android.content.Context;

import com.google.firebase.perf.metrics.AddTrace;
import com.song.deviceinfo.utils.DeviceUtils;

import java.util.ArrayList;
import java.util.List;

import androidx.core.util.Pair;

/**
 * Created by chensongsong on 2020/8/3.
 */
public class DeviceInfo {

    @AddTrace(name = "DeviceInfo.getDeviceInfo")
    public static List<Pair<String, String>> getDeviceInfo(Context context) {
        List<Pair<String, String>> list = new ArrayList<>();
        list.add(new Pair<>("AndroidId", DeviceUtils.getAndroidId(context)));
        list.add(new Pair<>("IMEI", DeviceUtils.getIMEI(context)));
        DeviceUtils.getDeviceInfo(context,list);
        list.add(new Pair<>("ICCID", DeviceUtils.getIccId(context)));
        DeviceUtils.getSimInfo(context,list);
        DeviceUtils.getOtherInfo(context,list);
        return list;
    }

}

File: app/src/main/java/com/song/deviceinfo/info/DeviceInfo.kt
package com.song.deviceinfo.info

import android.content.Context
import androidx.core.util.Pair
import com.google.firebase.perf.metrics.AddTrace
import com.song.deviceinfo.utils.DeviceUtils.getAndroidId
import com.song.deviceinfo.utils.DeviceUtils.getDeviceInfo
import com.song.deviceinfo.utils.DeviceUtils.getIMEI
import com.song.deviceinfo.utils.DeviceUtils.getIccId
import com.song.deviceinfo.utils.DeviceUtils.getOtherInfo
import com.song.deviceinfo.utils.DeviceUtils.getSimInfo

/**
 * Created by chensongsong on 2020/8/3.
 */
object DeviceInfo {
    @AddTrace(name = "DeviceInfo.getDeviceInfo")
    fun getDeviceInfo(context: Context): List<Pair<String, String>> {
        val list: MutableList<Pair<String, String>> = ArrayList()
        list.add(Pair("AndroidId", getAndroidId(context)))
        list.add(Pair("IMEI", getIMEI(context)))
        getDeviceInfo(context, list)
        list.add(Pair("ICCID", getIccId(context)))
        getSimInfo(context, list)
        getOtherInfo(context, list)
        return list
    }
}
